tilavarauspalvelu/utils/date_util.py

```diff
@@ -29,3 +29,16 @@ def previous_or_current_matching_weekday(
     if days_ahead > 0:  # Target day already happened this week
         days_ahead -= 7
     return d + datetime.timedelta(days_ahead)
+
+
+def localized_short_weekday(weekday: int, lang_code: str) -> str:
```

Comment:
This could also could make use of django's translations; https://docs.djangoproject.com/en/3.2/topics/i18n/translation/

Then we'd need to accept that the English abbreviation would be Mon, Tue, Wed etc.. (unless we provide custom translations files and maintain those).

So lets say we want "Mon" to be in reservee language => ``translation.activate(lang_code)`` and then ``return _("Mon")`` (that can probably get from the date library) where the ``_`` is imported from ``from django.utils.translation import gettext_lazy as _``

Reply:
Hmm, `translation.activate` applies the language selection to the current thread. I wonder if this could raise some "surprising features" 🤔 

Do you have any deeper understanding on how Django uses threads? I'm mostly thinking about the risk of setting the language and then serving requests in completely wrong language for other users.

Reply:
I think with current setup that we have one thread setupped in uwsgi conf it should be fine.

Reply:
oh, I backup a little => one thread but multiple processes so it might be funky ![](https://media.giphy.com/media/h0TPbr80IwoP6/giphy.gif)